editor/components/page-attributes/parent.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import { stringify } from 'querystringify';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withInstanceId, withAPIData } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+import { buildTermsTree } from '@wordpress/utils';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPostId, getEditedPostAttribute, getCurrentPostType } from '../../store/selectors';
+import { editPost } from '../../store/actions';
+import { TermTreeSelect } from '@wordpress/blocks';
+
+export function PageAttributesParent( { parent, postId, postType, items, onUpdateParent } ) {
+	const isHierarchical = get( postType, 'data.hierarchical', false );
+	if ( ! isHierarchical ) {
+		return null;
+	}
+	const parentPageLabel = get( postType, 'data.labels.parent_item_colon' );
+	const pagesTree = buildTermsTree( get( items, 'data', [] ).filter( item => item.id !== postId ).map( ( item ) => ( {
+		id: item.id,
+		parent: item.parent,
+		name: item.title.raw ? item.title.raw : `#${ item.id } (${ __( 'no title' ) })`,
```

Comment:
Do we want to show the id if no title? Is this how it works in the classic editor?

Reply:
Yes, I ported the classic editor behavior:)